sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -182,8 +186,8 @@ public FileBasedSink(String baseOutputFilename, String extension, String fileNam
    *
    * <p>See {@link ShardNameTemplate} for a description of file naming templates.
    */
-  public FileBasedSink(String baseOutputFilename, String extension, String fileNamingTemplate,
-      WritableByteChannelFactory writableByteChannelFactory) {
+  public FileBasedSink(ValueProvider<String> baseOutputFilename, String extension,
+      String fileNamingTemplate, WritableByteChannelFactory writableByteChannelFactory) {
```

Comment:
My take was that the canonical use case is "write files to a new directory every day", or similar.  Changing the suffix and fileTemplate seemed less valuable for that reason.